src/cpu/minor/ticked.cc

```diff
@@ -47,39 +47,29 @@ Ticked::Ticked(ClockedObject &object_,
     event([this]{ processClockEvent(); }, object_.name(), false, priority),
     running(false),
     lastStopped(0),
-    /* Allocate numCycles if an external stat wasn't passed in */
-    numCyclesLocal((imported_num_cycles ? NULL : new Stats::Scalar)),
-    numCycles((imported_num_cycles ? *imported_num_cycles :
-        *numCyclesLocal))
+    // Pass the external cycles stat to numCycles.
+    numCycles(*imported_num_cycles),
+    stats(this)
 { }
 
 void
 Ticked::processClockEvent() {
-    ++tickCycles;
+    ++stats.tickCycles;
     ++numCycles;
     countCycles(Cycles(1));
     evaluate();
     if (running)
         object.schedule(event, object.clockEdge(Cycles(1)));
 }
 
-void
-Ticked::regStats()
+Ticked::TickedStats::TickedStats(Stats::Group *parent)
+    : Stats::Group(parent, "Ticked"),
+      ADD_STAT(tickCycles,
+                    "Number of cycles that the object actually ticked."),
+      ADD_STAT(idleCycles, "Total number of cycles that the object"
+                    "has spent stopped.",numCycles - tickCycles)
```

Comment:
This is causing a compiler error. You can't use `numCycles` which is not a member of `TickedStats`.

```suggestion
Ticked::TickedStats::TickedStats(Ticked *parent)
    : Stats::Group(parent, "Ticked"),
      ADD_STAT(tickCycles,
                    "Number of cycles that the object actually ticked."),
      ADD_STAT(idleCycles, "Total number of cycles that the object"
                    "has spent stopped.", parent->numCycles - tickCycles)
```